kaitai/stream_test.go

```diff
@@ -74,6 +75,52 @@ func TestStream_Size(t *testing.T) {
 	}
 }
 
+type artificialError struct{}
+
+func (e artificialError) Error() string {
+	return "artificial error when seeking with io.SeekCurrent after seeking to end"
+}
+
+type failingReader struct {
+	pos int64
+}
+
+func (fr *failingReader) Read(p []byte) (n int, err error) { return 0, nil }
+func (fr *failingReader) Seek(offset int64, whence int) (int64, error) {
+	switch {
+	case whence == io.SeekCurrent && fr.pos == -1:
+		return 0, artificialError{} // returns an artificial error to simulate seeking error
+	case whence == io.SeekCurrent:
+		return fr.pos, nil
+	case whence == io.SeekStart:
+		fr.pos = offset
+	default:
```

Comment:
A minor thing: I'd make it explicit here that we primarily intend this case for `io.SeekEnd`:

```suggestion
	default: // whence == io.SeekEnd
```

Keeping a `default` there is probably a good thing, so let's just add that comment.